frame/contracts/src/lib.rs

```diff
@@ -867,8 +858,10 @@ decl_event! {
 
 decl_storage! {
 	trait Store for Module<T: Trait> as Contract {
-		/// Gas spent so far in this block.
-		GasSpent get(fn gas_spent): Gas;
+		/// The amount of gas left from the execution of the latest contract.
+		///
+		/// This value is transient and removed before block finalization.
```

Comment:
It's removed sooner, right? We can say it is removed immediately after runtime call dispatch (by the signed extension).

Reply:
Question about such transient values: 

Since they are never stored on chain, hence never used in consensus and state root, why do we even bother storing them on the same trie which imposes all the hashing overhead? Or are they not imposing any overhead in such uses?

Reply:
the overhead depend on the implementation of the client but currently our client has a cache-overlay-changes, so this value will be write in this cache, and the removed from this cache. at the end of the block build, this cache-overlay-changes will be committed without those values

Maybe we can find a solution not to call the sr-io api and just put that in the wasm memory but I don't know.

Reply:
Yeah, avoiding hitting the storage layer would be good anyway.

Some of the transient values have to survive through a runtime call boundary so they cannot be stored in memory.